Makefile

```diff
@@ -223,61 +247,100 @@ dist: dist-agent dist-agentctl dist-packages
 	pushd dist && sha256sum * > SHA256SUMS && popd
 .PHONY: dist
 
+####################
+# BEGIN AGENT DIST #
+####################
+
 dist-agent: seego dist/agent-linux-amd64 dist/agent-linux-arm64 dist/agent-linux-armv6 dist/agent-linux-armv7 dist/agent-darwin-amd64 dist/agent-darwin-arm64 dist/agent-windows-amd64.exe dist/agent-freebsd-amd64 dist/agent-windows-installer.exe
 dist/agent-linux-amd64: seego
-	@CGO_ENABLED=1 GOOS=linux GOARCH=amd64; $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+	$(call SetBuildVarsConditional,linux/amd64) ;      $(seego) build $(CGO_FLAGS)  -o $@ ./cmd/agent
+
 dist/agent-linux-arm64: seego
-	@CGO_ENABLED=1 GOOS=linux GOARCH=arm64; $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+	$(call SetBuildVarsConditional,linux/arm64) ;      $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+
 dist/agent-linux-armv6: seego
-	@CGO_ENABLED=1 GOOS=linux GOARCH=arm GOARM=6; $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+	$(call SetBuildVarsConditional,linux/arm/v6) ;      $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+
 dist/agent-linux-armv7: seego
-	@CGO_ENABLED=1 GOOS=linux GOARCH=arm GOARM=7; $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+	$(call SetBuildVarsConditional,linux/arm/v7) ;      $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+
 dist/agent-linux-mipsle: seego
-	@CGO_ENABLED=1 GOOS=linux GOARCH=mipsle GOMIPS=softfloat; $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
-dist/agent-darwin-amd64: seego
-	@CGO_ENABLED=1 GOOS=darwin GOARCH=amd64; $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+	$(call SetBuildVarsConditional,linux/mipsle) ;      $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+
+dist/agent-darwin-amd64:  seego
+	$(call SetBuildVarsConditional,darwin/amd64) ;      $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+
 dist/agent-darwin-arm64: seego
-	@CGO_ENABLED=1 GOOS=darwin GOARCH=arm64; $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+	$(call SetBuildVarsConditional,darwin/arm64) ;      $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+
 dist/agent-windows-amd64.exe: seego
-	@CGO_ENABLED=1 GOOS=windows GOARCH=amd64; $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+	$(call SetBuildVarsConditional,windows) ;      $(seego) build $(CGO_FLAGS) -o $@ ./cmd/agent
+
 dist/agent-windows-installer.exe: dist/agent-windows-amd64.exe
-	cp dist/agent-windows-amd64.exe ./packaging/windows
-	cp LICENSE ./packaging/windows
-	docker build ./packaging/windows -t windows_nsis
-	docker run -e VERSION=${RELEASE_TAG} --rm -t -v $(CURDIR)/dist:/app windows_nsis
+ifeq ($(DRONE),true)
+	cp dist/agent-windows-amd64.exe ./packaging/windows && cp LICENSE ./packaging/windows && makensis -V4 -DVERSION=${RELEASE_TAG} -DOUT="../../dist/grafana-agent-installer.exe" ./packaging/windows/install_script.nsis
```

Comment:
I noticed the DOUT path is different from the else on line 289, can you help me understand what's going on there?

Reply:
Fixed this and tested on windows native and dist/containers.